File: src/clock.zig
const c = @import("c.zig");

pub const Clock = struct {
    const Self = @This();
    current_frame: u64,
    last_frame: u64,

    pub fn init() Self {
        return Self{
            .current_frame = 0,
            .last_frame = 0,
        };
    }

    pub fn deltaTime(self: *Self) u64 {
        self.current_frame = c.SDL_GetTicks64();
        var dt = self.current_frame - self.last_frame;
        self.last_frame = self.current_frame;

        return dt;
    }
};

File: src/clock.cpp
#include "clock.h"
#include <SDL2/SDL.h>

Clock::Clock() {
	last_frame = 0.0f;
}

double Clock::delta_time() {
	double current_frame = SDL_GetTicks64();
	double delta_time = current_frame - last_frame;
	last_frame = current_frame;

	return delta_time;
}
